folium/map.py

```diff
@@ -402,9 +401,7 @@ def __init__(
             draggable=draggable or None, autoPan=draggable or None, **kwargs
         )
         if icon is not None:
-            # this makes sure it is added only once
-            self._parent.add_child(icon, name=icon.get_name(), index=0)
-            self.icon = icon
+            self.add_child(icon)
```

Comment:
Why did the above code not work?

Reply:
It worked I think, but it's not necessary. `Figure.script` is an OrderedDict, so we are already guaranteed that multiple renders of the same Icon object will result in only one entry in `Figure.script`.